### PR TITLE
Delete Users

### PR DESCRIPTION
**Pull request by shrutiyer**
Only admins can delete users. They cannot directly delete other admins. In order to do that, they will have to change the admin status and then delete them

### CODE REVIEW

**Comment by aconsilvio** [comment]
👍

**Comment by BrennaManning** [comment]
:+1:

File: routes/userRoutes.js [hunk 1]
```diff
@@ -155,4 +155,18 @@ routes.POSTchangePassword = function(req, res) {
   });
 };
 
+routes.POSTdelUser = function(req, res) {
+  var userid = req.params._id;
+  console.log('userid: '+userid);
```
Comment (davidabrahams) [inline routes/userRoutes.js]: space

Comment (davidabrahams) [inline routes/userRoutes.js]: actually, just remove this

File: public/javascripts/settingsController.js [hunk 2]
```diff
@@ -14,6 +14,14 @@ var settingsController = function($scope, $http, $location) {
     });
   };
 
+  $scope.delUser = function(username, userArray, delUserIdx) {
+    $http.post('api/delUser/'+username._id).then(function successCallback(response) {
+      userArray.splice(delUserIdx, 1);
```
Comment (davidabrahams) [inline public/javascripts/settingsController.js:19]: why the splice?

File: routes/index.js [hunk 3]
```diff
@@ -20,6 +20,7 @@ routes.GETemailver = userRoutes.GETemailver;
 routes.GETallUsers = userRoutes.GETallUsers;
 routes.POSTchangeAdmin = userRoutes.POSTchangeAdmin;
 routes.POSTchangePassword = userRoutes.POSTchangePassword;
+routes.POSTdelUser = userRoutes.POSTdelUser;
```
Comment (davidabrahams) [inline routes/index.js]: You can actually make HTTP `DELETE` requests. Maybe this would be proper here? i have no idea

**Comment by davidabrahams** [comment]
had a few comments, otherwise :+1:

File: public/javascripts/settingsController.js (hunk 2 above)
Comment (shrutiyer) [inline public/javascripts/settingsController.js:19]: On success, it deletes the clicked row from usernames array

File: routes/index.js (hunk 3 above)
Comment (shrutiyer) [inline routes/index.js]: Good point

**Comment by shrutiyer** [comment]
Cool. Made changes. 
@nmohamed?

File: public/javascripts/settingsController.js (hunk 2 above)
Comment (davidabrahams) [inline public/javascripts/settingsController.js:19]: It seems like a better solution is to have the route return all the users in the `response`, and then to update the users to the list of users in the response.

Comment (shrutiyer) [inline public/javascripts/settingsController.js:19]: The callback for findOneAndRemove returns the deleteled object. We'll have to make another GET request to get all the remaining users in the callback. I think doing splice is nicer.

Comment (davidabrahams) [inline public/javascripts/settingsController.js:19]: agreed. merging!

**Pull request merged**